### PR TITLE
Remove helanal

### PR DESCRIPTION
**Pull request by lilyminium**
Fixes #

Changes made in this Pull Request:
 - remove old helanal module

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [x] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/analysis/__init__.py [hunk 1]
```diff
@@ -56,7 +56,7 @@
     Analyze hydrogen bonds, including both the per frame results as well
     as the dynamic properties and lifetimes.
 
-:mod:`~MDAnalysis.analysis.helanal`
+:mod:`~MDAnalysis.analysis.helix_analysis`
```
Comment (IAlibay) [inline package/MDAnalysis/analysis/__init__.py:59]: Probably will want to remove the entry from `all` below?
Also the sphinx stub for helanal, and the entry in analysis_modules.rst.

**Review by IAlibay — CHANGES_REQUESTED** [review]
Thanks for doing this :)

Can you also add a change under the changelog's `changes`?

File: package/MDAnalysis/analysis/__init__.py (hunk 1 above)
Comment (lilyminium) [reply package/MDAnalysis/analysis/__init__.py:59]: Thanks @IAlibay!

**Comment by IAlibay** [comment]
@lilyminium 

`/home/travis/build/MDAnalysis/mdanalysis/package/MDAnalysis/analysis/helix_analysis.py:docstring of MDAnalysis.analysis.helix_analysis:10:citation not found: Bansal2000`

I guess the citation needs to be ported over.

There's also some flaky lib_dcd failure on azure, it's not the first time I've seen it, not sure what's going on there.

**Comment by lilyminium** [comment]
Thanks for looking at this @IAlibay!

**Review by IAlibay — APPROVED** [review]

**Pull request merged**